### PR TITLE
added loop_until statement into tests

### PR DESCRIPTION
**Pull request by jstavel**
I read all tests and I updated them to use loop_until statement when it makes sense.
I have tested it and fixed a few errors.

### CODE REVIEW

File: integration-tests/test_file_workflow.py [hunk 1]
```diff
@@ -46,12 +49,19 @@ def test_file_workflow_with_an_archive_with_only_one_canonical_fact(
     # once archive is uploaded system status changes to register
     assert conftest.loop_until(lambda: insights_client.is_registered)
 
-    insights_client.run("--check-results")  # to get host details from inventory
-    with open(HOST_DETAILS, "r") as data_file:
-        file_content = json.load(data_file)
-        host_data = file_content["results"][0]
-    assert host_data["insights_id"] == machine_id
-    assert host_data["fqdn"] is None
+    host_data = {}
```
Comment (Archana-PandeyM) [inline integration-tests/test_file_workflow.py:52]: No need to combine these changes here, we have already implemented API get/post in pytest_client_tools, so if needed refactoring we will do it in a separate PR for all the --check-results occurrences.

**Review by Archana-PandeyM — CHANGES_REQUESTED** [review]

File: integration-tests/test_file_workflow.py [hunk 2]
```diff
@@ -4,6 +4,9 @@
 from constants import HOST_DETAILS
 from constants import MACHINE_ID_FILE
 import conftest
+import logging
```
Comment (Archana-PandeyM) [inline integration-tests/test_file_workflow.py:7]: we can replace it with pytest_client_tools logged_run

**Review by Archana-PandeyM — CHANGES_REQUESTED** [review]

**Comment by Archana-PandeyM** [comment]
missed this iteration, we should add a check here also https://github.com/RedHatInsights/insights-client/pull/266/files#diff-e37f0bdad557c241a3aa21f88425bd1a32acf0eb499fc2ce37371de5fa7ff2f5L160

**Comment by Archana-PandeyM** [comment]
Missed changes for test_motd.py

**Comment by Archana-PandeyM** [comment]
few more in test_client L160 L 64 etc.

File: integration-tests/test_file_workflow.py (hunk 1 above)
Comment (jstavel) [reply integration-tests/test_file_workflow.py:52]: Yes, I will refactor it in a separate PR

File: integration-tests/test_file_workflow.py (hunk 2 above)
Comment (jstavel) [reply integration-tests/test_file_workflow.py:7]: I will replace it in my next PR where I will rewrite usage of --check-results command to get inventory data.

**Comment by jstavel** [comment]
Hi Archana. Thanks for your review. Regarding to missing loop_statements in test_registration, ...
- I have tried to understand what a test is about. IE. if a test verifies behaviour that does not depend on response from console service - it is not necessary to wait for the response. For example - a test verifies that log files appear in log directory. It is not necessary to wait for a response - the application should do some action even the response is not returned. So I wanted to save system time not applying loop_until statement after every 'insights_client.register()'.

On the other hand - your suggestion for test_client.py L160 makes sense. So I added the loop_statement there.

File: integration-tests/test_file_workflow.py [hunk 3]
```diff
@@ -46,12 +49,19 @@ def test_file_workflow_with_an_archive_with_only_one_canonical_fact(
...
 
-    insights_client.run("--check-results")  # to get host details from inventory
-    with open(HOST_DETAILS, "r") as data_file:
-        file_content = json.load(data_file)
-        host_data = file_content["results"][0]
-    assert host_data["insights_id"] == machine_id
-    assert host_data["fqdn"] is None
+    host_data = {}
+
+    def actual_machine_id_appears_in_inventory():
```
Comment (Archana-PandeyM) [inline integration-tests/test_file_workflow.py:54]: Please move it to seperate PR

**Review by Archana-PandeyM — CHANGES_REQUESTED** [review]

**Comment by Archana-PandeyM** [comment]
other changes look good, Thanks for the PR Jan,
Please clean merge commits /squash them.

File: integration-tests/test_file_workflow.py [hunk 4]
```diff
@@ -5,6 +5,7 @@
 from constants import MACHINE_ID_FILE
 import conftest
 
+
```
Comment (ptoscano) [inline integration-tests/test_file_workflow.py:8]: extra newline addition

File: integration-tests/test_file_workflow.py [hunk 5]
```diff
@@ -45,13 +46,14 @@ def test_file_workflow_with_an_archive_with_only_one_canonical_fact(
 
     # once archive is uploaded system status changes to register
     assert conftest.loop_until(lambda: insights_client.is_registered)
-
+
```
Comment (ptoscano) [inline integration-tests/test_file_workflow.py:49]: extra whitespace changes (this also breaks `black`)

File: integration-tests/test_file_workflow.py [hunk 6]
```diff
@@ -45,13 +46,14 @@ def test_file_workflow_with_an_archive_with_only_one_canonical_fact(
 
     # once archive is uploaded system status changes to register
     assert conftest.loop_until(lambda: insights_client.is_registered)
-
+    
     insights_client.run("--check-results")  # to get host details from inventory
     with open(HOST_DETAILS, "r") as data_file:
         file_content = json.load(data_file)
         host_data = file_content["results"][0]
+
```
Comment (ptoscano) [inline integration-tests/test_file_workflow.py:54]: extra newline addition

File: integration-tests/test_file_workflow.py [hunk 7]
```diff
@@ -45,13 +46,14 @@ def test_file_workflow_with_an_archive_with_only_one_canonical_fact(
...
     assert host_data["insights_id"] == machine_id
-    assert host_data["fqdn"] is None
+    assert "fqdn" not in host_data
```
Comment (ptoscano) [inline integration-tests/test_file_workflow.py:56]: unrelated change, please split it in its own PR

**Review by ptoscano — CHANGES_REQUESTED** [review]
please squash the commits into one with a proper commit message explaining all the changes
